src/main/java/atmosfears/AtmosFearsBE/controller/DataEndpointController.java

```diff
@@ -0,0 +1,39 @@
+package atmosfears.AtmosFearsBE.controller;
+
+import atmosfears.AtmosFearsBE.service.AirParticulatesService;
+import org.json.JSONObject;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+
+@RestController
+public class DataEndpointController {
+
+    private AirParticulatesService airParticulatesService;
+
+    public DataEndpointController(AirParticulatesService airParticulatesService) {
+        this.airParticulatesService = airParticulatesService;
+    }
+
+
+    @GetMapping("/data/average")
+    public ResponseEntity<Map<String, Object>> getAverageParticulates(@RequestParam(name = "start") String startDateStr,
+                                                      @RequestParam(name = "end") String endDateStr){
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd-hh:mm");
+        Date startDate, endDate;
+        try {
+            startDate = simpleDateFormat.parse(startDateStr);
+            endDate = simpleDateFormat.parse(endDateStr);
+        } catch (ParseException e) {
+            e.printStackTrace();
+            return ResponseEntity.internalServerError().build();
```

Comment:
Return bad request response instead of throwing an internal server error. Internal server error means that something unexpected happened on the server and it was not intended (e.g. some developer mistake). If user provides wrong date formats in the request, the request is invalid and you should return a descriptive response with information what is incorrect and the proper status code (400 - bad request).